File: src/lib/logger.ts
import chalk from "chalk";

export const logger = {
  info(msg: string): void {
    process.stdout.write(`${chalk.blue("info")} ${msg}\n`);
  },
  success(msg: string): void {
    process.stdout.write(`${chalk.green("success")} ${msg}\n`);
  },
  warn(msg: string): void {
    process.stderr.write(`${chalk.yellow("warn")} ${msg}\n`);
  },
  error(msg: string): void {
    process.stderr.write(`${chalk.red("error")} ${msg}\n`);
  },
};

File: internal/logger/logger.go
package logger

import (
	"fmt"
	"os"

	"github.com/fatih/color"
)

var (
	blue   = color.New(color.FgBlue).SprintFunc()
	green  = color.New(color.FgGreen).SprintFunc()
	yellow = color.New(color.FgYellow).SprintFunc()
	red    = color.New(color.FgRed).SprintFunc()
)

func Info(format string, a ...any) {
	fmt.Fprintf(os.Stdout, "%s %s\n", blue("info"), fmt.Sprintf(format, a...))
}

func Success(format string, a ...any) {
	fmt.Fprintf(os.Stdout, "%s %s\n", green("success"), fmt.Sprintf(format, a...))
}

func Warn(format string, a ...any) {
	fmt.Fprintf(os.Stderr, "%s %s\n", yellow("warn"), fmt.Sprintf(format, a...))
}

func Error(format string, a ...any) {
	fmt.Fprintf(os.Stderr, "%s %s\n", red("error"), fmt.Sprintf(format, a...))
}
